lib/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] Please define the relationship between the Decorator and Nameable, it has to be done by using the accessor's attributes such as `attr_accessor :nameable`. 🙏